### ISSUE TITLE
how to use controlnet on flux

### ISSUE THREAD

**Issue by chaobeiying** [lllyasviel/stable-diffusion-webui-forge#1685]
I use 'flux1-schnell.safetensors  ' at models/Stable-diffusion    and use 'flux-canny-controlnet.safetensors' at modes/ControlNet  
but when I run txt2img ,it occurs  error as :ERROR - Recognizing Control Model failed: /AISD/stable-diffusion-webui-forge/models/ControlNet/flux-canny-controlnet.safetensors
AssertionError: None
![微信截图_20240904153138](https://github.com/user-attachments/assets/61d8aa06-c780-420a-b2ac-0afb99191923)

thank you!

**Comment by wcole3** [comment]
Flux controlnet not yet implemented, WiP

**Comment by Vagrin** [comment]
> Flux controlnet not yet implemented, WiP

That's great to know! If something is working for everyone else then it's worthwhile figuring out why it won't work on your own machine. If it's working for no one? Fair enough. We wait patiently... with bated breath.

**Comment by bolli20000** [comment]
Still Looking also for an working controlnet update after 2 weeks. Any news to this topic ?

**Comment by Vagrin** [comment]
> Still Looking also for an working controlnet update after 2 weeks. Any news to this topic ?

This was pointed out on a post:
Update Sep 1:

The rewrite of ControlNet Intergrated will start at about Sep 29. The estimated finish date is about Oct 7. When this note is announced, the main targets include some diffusers formatted Flux ControlNets and some community implementation of Union ControlNets. However, this may be extended if stronger models come out after this note.

From this post:
https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/932

So... we'll get there. You get what you pay for!

**Comment by alebeard** [comment]
Somewhat off topic but not sure where else to ask Does anyone have good
control net models that we'll work with PONYXL And if so could you point me
to a link.

**Comment by Vagrin** [comment]
> Somewhat off topic but not sure where else to ask Does anyone have good control net models that we'll work with PONYXL And if so could you point me to a link.
> […](#)
Every ControlNet and even ADetailer that worked in SDXL worked in Pony for me. If they're not working in Pony for you something might be broken. If you're trying to use SD1.5 ControlNets in Pony, I can't help you. I left SD1.5 a long time ago. It was great when it was all we had.

**Comment by alebeard** [comment]
Its not that I am trying to use sd1.5 its that the sdxl ones all seem to
such, found mistoline which works good, But all the rest of the models have
seemed to not work particularly well. Also there's a third window over to
the right that I have no idea what does that is fairly new, I was just
wondering if anyone had a link to some models that they knew were working
well as civitai does not seem to have a full suite of sdxl models Unless I
am missing them due to user error which is a distinct possibility.

**Comment by dan4ik94** [comment]
@alebeard the xinsir ones work well. https://huggingface.co/xinsir
If you want a bundle of many controlnets in one file I can recommended the **union** or **promax** model.

**Comment by XeonG** [comment]
> @alebeard the xinsir ones work well. https://huggingface.co/xinsir If you want a bundle of many controlnets in one file I can recommend the **union** or **promax** model.

any of that work on forge?

**Comment by dan4ik94** [comment]
> any of that work on forge?

they work fine, I'm using them every day.

**Comment by Vagrin** [comment]
> > any of that work on forge?
> 
> they work fine, I'm using them every day.

Holy shit! That's news to me and a bunch of other people too I'd guess. Do you just use them like you would with Pony or SDXL? Or do you have to do something special to get them to work? Please. Feel free to inform us.

**Comment by ronbere** [comment]
> > any of that work on forge?
> 
> they work fine, I'm using them every day.

Not working on forge....

**Comment by dan4ik94** [comment]
> Holy shit! That's news to me and a bunch of other people too I'd guess. Do you just use them like you would with Pony or SDXL? Or do you have to do something special to get them to work? Please. Feel free to inform us.

use what, I was talking about sdxl/pony not flux when responding to the guy.

**Comment by Vagrin** [comment]
> > Holy shit! That's news to me and a bunch of other people too I'd guess. Do you just use them like you would with Pony or SDXL? Or do you have to do something special to get them to work? Please. Feel free to inform us.
> 
> use what, I was talking about sdxl/pony not flux when responding to the guy.

Fair enough. Didn't read through the entire thing you initially responded to. You can imagine how excited I was!

**Comment by dan4ik94** [comment]
> Fair enough. Didn't read through the entire thing you initially responded to. You can imagine how excited I was!

we'll have it next week probably so let's wait

**Comment by SomeNiceDude** [comment]
Would be nice if this issue has been acknowledged at least and is being worked on

**Comment by tommywright** [comment]
Any update on this? I'd love to have a working controlnet for flux in forge. :)

**Comment by Sohlis-afk** [comment]
Any update?
